hax/hax/util.py

```diff
@@ -207,6 +207,28 @@ def to_payload(v: TxPutKV) -> Dict[str, Any]:
         except (ConsulException, HTTPError, RequestException) as e:
             raise HAConsistencyException('Failed to put value to KV') from e
 
+    @repeat_if_fails(max_retries=5)
+    def kv_delete_in_transaction(self, tx_payload: Dict[str, str]) -> bool:
```

Comment:
Can we use an explicit type for `tx_payload` instead of raw `Dict[str, str]`?

The problem is that it is too easy to provide a dict with incorrect structure that your function will fail to process. If you use `@dataclass` or `NamedTuple`, the requirements to such a dict become explicit and can be found during static analysis.

Reply:
Done